extensions/ql-vscode/src/extension.ts

```diff
@@ -431,13 +431,16 @@ export async function activate(
         return;
       }
 
-      if (CliVersionConstraint.OLDEST_SUPPORTED_CLI_VERSION.compare(ver) < 0) {
+      if (
+        CliVersionConstraint.OLDEST_SUPPORTED_CLI_VERSION.compare(ver.version) <
+        0
+      ) {
         return;
       }
 
       void showAndLogWarningMessage(
         extLogger,
-        `You are using an unsupported version of the CodeQL CLI (${ver}). ` +
+        `You are using an unsupported version of the CodeQL CLI (${ver.version}). ` +
```

Comment:
How confident are you you've found all of these places where we insert the version into a string? I haven't exhaustively gone through trying to find them, so I just thought I'd  ask how you feel.

If we miss any it'll lead to a classic `[Object object]` which wouldn't look too good.

Reply:
I feel pretty confident. The only place the `VersionAndFeatures` gets exposed outside of `CodeQLCliServer` is via the version changed listener event (here). I made `CodeQLCliServer.getVersionAndFeatures` private, so anyone else will either call `getVersion` or `getFeatures`.